erpnext/selling/doctype/sales_order/test_sales_order.py

```diff
@@ -86,7 +86,7 @@ def create_dn_against_so(self, so, delivered_qty=0):
 		dn.get("delivery_note_details")[0].against_sales_order = so.name
 		dn.get("delivery_note_details")[0].prevdoc_detail_docname = so.get("sales_order_details")[0].name
 		if delivered_qty:
-			dn.get("delivery_note_details")[0].qty = delivered_qty
+			dn.get("delivery_note_details")[0].stock_qty = delivered_qty
```

Comment:
As we are calculating `stock_qty` on saving of the document, we have to assign `qty` and `conversion_factor`, otherwise `stock_qty` will be set as zero.